MetaMask/metamask-mobile
# chore(3663): removing the isMultichainVersion1Enabled feature flag, and corresponding dead code

## **Description**
- Step 1 **(done)**: remove code that ran in the feature flag OFF scenario, confirm stuff runs as expected (manually and e2e)
- Step 2 **(done)**: remove any left over checks on the feature flags, for examples checks that used only feature flag ON to render something specific to that scenario
- Step 3 **(done)**: address unit test snapshots tests failing, all the while using these snapshots to ensure expected behavior is maintained, comparing a test that runs with FF OFF, to the existing snapshot on main, this test should pass since FF OFF is the default, this a baseline. Next, compare the snapshot generated from a FF ON test running on the code with feature flag, to the snapshot generated once all the dead code and feature flag code is removed, that test should pass since FF ON becomes the default behavior once feature flag and dead code is removed.
- Step 4 **(done)**: address unit tests that fail on things other than snapshots, find out why they are failing
- Step 5 **(done)**: added coverage on new code (Sonar)



## **Related issues**

Contributes to: https://github.com/MetaMask/MetaMask-planning/issues/3663

## **Manual testing steps**

1. Go to the in app browser and test-dapp
2. Connect to it
3. Click the top header in the in-app browser, confirm the expected permissions as displaying.

## **Screenshots/Recordings**



### **Before**

It's a refactoring, to the user it looks the same before and after.

### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
